wallet/src/signer/trezor_signer/mod.rs

```diff
@@ -0,0 +1,1114 @@
+                            }
+                        }
+                        multisig_partial_signature::SigsVerifyResult::Invalid => {
+                            unreachable!(
+                                "We checked the signatures then added a signature, so this should be unreachable"
+                            )
+                        }
+                    };
+
+                    let sig = InputWitness::Standard(StandardInputSignature::new(
+                        sighash_type,
+                        current_signatures.encode(),
+                    ));
+                    return Ok((Some(sig), status));
+                }
+
+                Ok((None, SignatureStatus::NotSigned))
+            }
+            Destination::ScriptHash(_) => Ok((None, SignatureStatus::NotSigned)),
+        }
+    }
+
+    fn to_trezor_output_msgs(&self, ptx: &PartiallySignedTransaction) -> Vec<MintlayerTxOutput> {
+        let outputs = ptx
+            .tx()
+            .outputs()
+            .iter()
+            .map(|out| to_trezor_output_msg(&self.chain_config, out))
+            .collect();
+        outputs
+    }
+}
+
+impl Signer for TrezorSigner {
+    fn sign_tx(
+        &mut self,
+        ptx: PartiallySignedTransaction,
+        key_chain: &impl AccountKeyChains,
+        _db_tx: &impl WalletStorageReadUnlocked,
+    ) -> SignerResult<(
+        PartiallySignedTransaction,
+        Vec<SignatureStatus>,
+        Vec<SignatureStatus>,
+    )> {
+        let inputs = to_trezor_input_msgs(&ptx, key_chain, &self.chain_config)?;
+        let outputs = self.to_trezor_output_msgs(&ptx);
+        let utxos = to_trezor_utxo_msgs(&ptx, &self.chain_config);
+
+        let new_signatures = self
+            .client
+            .lock()
+            .expect("poisoned lock")
+            .mintlayer_sign_tx(inputs, outputs, utxos)
+            .map_err(TrezorError::DeviceError)?;
+
+        let inputs_utxo_refs: Vec<_> = ptx.input_utxos().iter().map(|u| u.as_ref()).collect();
+
+        let (witnesses, prev_statuses, new_statuses) = ptx
+            .witnesses()
+            .iter()
+            .enumerate()
+            .zip(ptx.destinations())
+            .map(|((i, witness), destination)| match witness {
+                Some(w) => match w {
+                    InputWitness::NoSignature(_) => Ok((
+                        Some(w.clone()),
+                        SignatureStatus::FullySigned,
+                        SignatureStatus::FullySigned,
+                    )),
+                    InputWitness::Standard(sig) => match destination {
+                        Some(destination) => {
+                            let sighash =
+                                signature_hash(sig.sighash_type(), ptx.tx(), &inputs_utxo_refs, i)?;
+
+                            if sig
+                                .verify_signature(&self.chain_config, destination, &sighash)
```

Comment:
`tx_verifier::input_check::signature_only_check::verify_tx_signature` should be used to verify htlc properly